src/extension.ts

```diff
@@ -163,16 +164,10 @@ export function activate(context: vscode.ExtensionContext): Promise<void> {
 
 			telemetryProvider.registerCommandWithTelemetry(
 				'vscode-tye.commands.debugAll',
-				async () => {
-					const applications = await tyeApplicationProvider.getApplications();
+				async (context, node: TyeApplicationNode) => {
+					const application = node.application;
 
-					// NOTE: We arbitrarily only attach to processes associated with the first application.
-					//       This matches the tree view, which also shows only that first application.
-					//       Future work will refactor the tree view and debugging for multiple applications
-					//       once Tye has better discovery support.
-					const application = applications[0];
-
-					if (application?.projectServices) {
+					if (node.application?.projectServices) {
```

Comment:
Nit: We can probably get away without the change on this line because `application` still has the value `node.application` assigned on line 168.

Reply:
Yes, I originally was going to eliminate the local `application`, but then changed my mind and forgot to revert that.